src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -87,6 +87,12 @@ public static bool IsStructMarshallingRequired(TypeDesc typeDesc)
             return !IsBlittableType(type);
         }
 
+        // Set of functions that need to be resolved statically in order for 
+        // other P/Invokes to use lazy resolution.
+        // In particular, these functions are necessary to get the current application
+        // path, so having them use lazy resolution would end up causing StackOverflow
+        private static readonly string[] s_lazyPInvokeDependencies = { "GetModuleFileNameW", "CoTaskMemAlloc", "CoTaskMemFree",
```

Comment:
I believe we need both:
- Allow specifying what to delay load vs. eager load/statically link on command line. It is important if you do not have a control over the library that is doing the PInvoke. Also, you may want to link the same library different ways in different projects.
- Allow marking individual PInvokes as eager load/static link. Mono has a prior art here: It treats __Internal as static link. I am not saying that it is a nice clean design, but we will need to support this hack if we ever want CoreRT to be reasonably compatible with Mono.